### PR TITLE
Wget to curl

### PR DESCRIPTION
**Pull request by joolswills**
Move RetroPie to use curl rather than wget.

This changeset switches the helpers download function to use curl (including some function enhancements such as stdout support) and makes downloadAndExtract also use the core download function.

In addition almost all module code using wget has been switched to use the download or downloadAndExtract function excluding one case where wget was used in an external script which has been switched to curl.

Packaging and setup functions have been adjusted to curl also.

This is left as draft as there's a few things I want to look at still, such as the stderr output capturing and some additional tweaks here and there perhaps. But feel free to test.

### CODE REVIEW

File: scriptmodules/packages.sh [hunk 1]
```diff
@@ -92,9 +92,9 @@ function rp_callModule() {
             rp_hasBinary "$md_id"
             local ret="$?"
 
-            # check if we had a network failure from wget
-            if [[ "$ret" -eq 4 ]]; then
-                __ERRMSGS+=("Unable to connect to the internet")
+            # check if we get a couldn't resolve host or failed to connec to host from curl
```
Comment (cmitu) [inline scriptmodules/packages.sh:95]: Small typo here (_connec_).

File: scriptmodules/supplementary/raspbiantools.sh [hunk 2]
```diff
@@ -77,7 +77,7 @@ function install_firmware_raspbiantools() {
     for pkg in "${pkgs[@]}"; do
         if hasPackage "$pkg" "$ver" ne; then
             deb="${pkg}_${ver}_armhf.deb"
-            if ! wget -O"$deb" "$url/$deb"; then
+            if download "$url/$deb"; then
```
Comment (cmitu) [inline scriptmodules/supplementary/raspbiantools.sh:80]: shouldn't this be `! download "$url/$deb"` ?

Comment (joolswills) [reply scriptmodules/supplementary/raspbiantools.sh:80]: Yes. Thanks.

File: scriptmodules/packages.sh (hunk 1 above)
Comment (joolswills) [reply scriptmodules/packages.sh]: Fixed

File: scriptmodules/supplementary/raspbiantools.sh (hunk 2 above)
Comment (joolswills) [reply scriptmodules/supplementary/raspbiantools.sh]: Fixed thanks.

**Comment by joolswills** [comment]
I have added timeout defaults which can be overridden and have reworked the error capturing.

This is ready now for review/testing.

**Comment by joolswills** [comment]
Actually I need to rebase it first due to a conflict with the spelling fixes.

**Comment by joolswills** [comment]
Apologies - not ready. Issue related to my stdout/stderr changes when piping output of download

**Comment by joolswills** [comment]
Reverted to the previous working redirection code. So ready for testing and review. Thanks.

**Comment by joolswills** [comment]
Going to merge. I have tested this a fair amount but will keep an eye if there are any related issues reported.

**Comment by joolswills** [comment]
I will run a full binary rebuild now it's merged also which should pick up any issues I have missed

**Pull request merged**